### ISSUE TITLE
Not working with menu_tags

### ISSUE THREAD

**Issue by fp4code** [divio/djangocms-snippet#35]
_[edit] need to remove from the title "(nor any classy-tags)"_

With a tag from menu_tags as show_breadcrumb
```
{% load menu_tags %}
{% show_breadcrumb %}
```
I got:
```
<type 'exceptions.ValueError'>
dictionary update sequence element #0 has length 4; 2 is required
```

By the way I've got the informational message by the help of https://github.com/divio/djangocms-snippet/pull/36

**Comment by fp4code** [comment]
Here is the traceback:
```
Traceback (most recent call last):
  File "./djangocms_snippet/cms_plugins.py", line 38, in render
    content = t.render(Context(context))
  File "python2.7/site-packages/django/template/base.py", line 208, in render
    return self._render(context)
  File "python2.7/site-packages/django/test/utils.py", line 94, in instrumented_test_render
    return self.nodelist.render(context)
  File "python2.7/site-packages/django/template/base.py", line 994, in render
    bit = node.render_annotated(context)
  File "python2.7/site-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
  File "python2.7/site-packages/classytags/core.py", line 153, in render
    return self.render_tag(context, **kwargs)
  File "python2.7/site-packages/classytags/helpers.py", line 88, in render_tag
    flatten_context(self.get_context(context, **kwargs))
  File "python2.7/site-packages/classytags/utils.py", line 106, in flatten_context
    return context.flatten()
  File "python2.7/site-packages/django/template/context.py", line 123, in flatten
    flat.update(d)
ValueError: dictionary update sequence element #0 has length 4; 2 is required
```

**Comment by fp4code** [comment]
I think there is illegal calls Context(context). Only legal is Context(dict).
Sorry, I'm no fluent with pull requests. This code solve my problem: https://github.com/fp4code/djangocms-snippet/commit/1cf5ae67ae78d0d9f7e38a0f8db446be9e1b0ae8

**Comment by FinalAngel** [comment]
@czpython any thoughts?

**Comment by fp4code** [comment]
The bug, step-by-step. No problem with Django==1.8, Buggy code since Django==1.9.

```bash
virtualenv -p /usr/bin/python3.5 env
source env/bin/activate
pip install djangocms-installer==0.9.7
djangocms mysite
cd mysite
./manage.py runserver
```

Important packages are:
```
Django==1.8.18
django-cms==3.4.4
djangocms-snippet==1.9.2
```
Add a Page http://localhost:8000/en/ and a Sub-Page http://localhost:8000/en/sub-page/

Add a Snippet "My Menu" 

```
{% load menu_tags %}
<p>Snippet menu:</p>
<ul>{% show_menu 0 100 100 100 %}</ul>
```

In the Placeholder Content of page http://localhost:8000/en/, add the Snippet "My Menu" and Publish

OK, we can see twice the menu. First the official menu, second a list "Page, Sub Page" after the text "Snippet menu:".

Now upgrade from Django==1.8.18 to 1.9.13:
```bash
pip install Django==1.9.13
```

migrate:
```bash
./manage.py migrate
```
start the server
```bash
./manage.py runserver
```
Now the snippet content has disappeared, with no message!

In order to see the error, apply this patch to ../env/lib/python3.5/site-packages/djangocms_snippet/cms_plugins.py:

```
--- ../env/lib/python3.5/site-packages/djangocms_snippet/cms_plugins.py.orig	2017-09-20 22:33:44.589894533 +0200
+++ ../env/lib/python3.5/site-packages/djangocms_snippet/cms_plugins.py	2017-09-20 22:52:14.001683345 +0200
@@ -4,6 +4,7 @@
 from django import template
 from django.conf import settings
 from django.template.context import Context
+from django.utils.html import escape
 from django.utils.safestring import mark_safe
 from django.utils.translation import ugettext_lazy as _
 
@@ -39,8 +40,9 @@
             content = _('Template %(template)s does not exist.') % {
                 'template': instance.snippet.template}
         except Exception:
-            exc = sys.exc_info()[0]
-            content = str(exc)
+            content = ('<pre>\n' +
+                       '\n'.join(map(lambda x: escape(str(x)), sys.exc_info())) +
+                       '\n<pre>')
         context.update({
             'content': mark_safe(content),
         })
```

start the server:
```bash
./manage.py runserver
```
Now the page http://localhost:8000/en/ is displaying an error:
```
<class 'ValueError'>
dictionary update sequence element #0 has length 9; 2 is required
```

**Comment by fp4code** [comment]
The `context` argument is build here: https://github.com/divio/django-cms/blob/release/3.4.x/cms/plugin_rendering.py#L397
```
context = PluginContext(context, instance, placeholder)
```
As shown here https://github.com/divio/django-cms/blob/release/3.4.x/cms/plugin_rendering.py#L596
that is a Context, not a dict.
Hence its illegal to call `Context(context)` here:
https://github.com/divio/djangocms-snippet/blob/1.9.2/djangocms_snippet/cms_plugins.py#L34
https://github.com/divio/djangocms-snippet/blob/1.9.2/djangocms_snippet/cms_plugins.py#L37

**Comment by fp4code** [comment]
Solved by pull request #41 
Keep in mind that djangocms_snippet/templatetags/snippet_tags.py may need a similar patch.
